include/zephyr/dt-bindings/gpio/st-stm32-gpio.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_GPIO_ST_STM32_GPIO_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_GPIO_ST_STM32_GPIO_H_
+
+/**
+ * @brief ST STM32 GPIO specific flags
+ *
+ * The driver flags are encoded in the 8 upper bits of @ref gpio_dt_flags_t as
+ * follows:
+ *
+ * - Bit 8: Configure a GPIO pin to power on the system after Poweroff.
+ *
+ * @ingroup gpio_interface
+ * @{
+ */
+
+/**
+ * Configures a GPIO pin to power on the system after Poweroff.
+ * This flag is reserved to GPIO pins that are associated with wake-up pins
+ * in STM32 PWR devicetree node, through the property "wkup-gpios".
+ */
+#define ST_STM32_GPIO_WKUP         (1 << 8)
```

Comment:
Please rename to STM32_GPIO... to stick with existing convention on flags naming